feat: Zotero 8 support

northword opened the pull request:
see also: https://groups.google.com/g/zotero-dev/c/uQhEGkJEzYs/m/6-pK0cvVBwAJ

- bump `strict_max_version` in `manifest.json` to `8.*`
- remove `Cu.unload` in `bootstrap.js` - it's not available in ff140
  > https://groups.google.com/g/zotero-dev/c/uQhEGkJEzYs/m/Hjqr13PwBwAJ
- bump deps (toolkit, types, scaffold)
- ~~bump esbuild target to `firefox140`~~
  Edited: We should not upgrade the target version at this time, as we may also need to be compatible with Zotero 7. After Zotero 8 becomes stable, we may be able to upgrade it.
- (optional) clean bootstrap.js: 
  - ~~remove register chrome - we can simply use `rootURI+relativePath` without registering `chrome://`~~
    > ~~In many cases, you may no longer need to register chrome:// URLs, as resources can be loaded by using a relative path directly or by appending a relative path to the rootURI string passed to your plugin's startup() function.~~
    > ~~https://www.zotero.org/support/dev/zotero_7_for_developers#chromemanifest_runtime_chrome_registration~~

    EDITED: reverted.
  - do not with Zotero init - startup only called after Zotero init
    > In Zotero 7, the install() and startup() bootstrap methods are called only after Zotero has initialized
    > https://www.zotero.org/support/dev/zotero_7_for_developers#xul_overlays_bootstrapjs
  - remove `rootURI = resourceURI.spec` - we have drop Zotero 6 compatibility 
    > https://www.zotero.org/support/dev/zotero_7_for_developers#xul_overlays_bootstrapjs

windingwind commented:
> remove register chrome - we can simply use rootURI+relativePath without registering chrome://

I don't remember, but in some cases, we probably still need to access the resources via a URI. Better to keep the registration here?

northword commented:
According to my tests, it seems that using `rooturi` is enough when loading plugin's resources. However, we can still keep this compatibility code here. I've already rebased and reverted the removal of `registerChrome`.

windingwind commented:
https://github.com/windingwind/zotero-plugin-template/actions/runs/16622587824/job/47030439793?pr=283
